### DISCUSSION TITLE
Excluding the model loading time, how long does inference take on T4 GPU?

### DISCUSSION THREAD

**Discussion by johnlee6511** [glucauze/sd-webui-faceswaplab#131] [General]
Hello!

I want to optimize inference time.
How long does inference take on T4 GPU for single image?

I takes 14 seconds excluding loading model.
There are any technique to reduce the time?

Thanks!

**Comment by glucauze** [comment]
Did you check GPU is really used and CUDA provider is selected ? It should be enabled on settings too. That's surprisingly slow. Without upscaling it should be less than 2s. It is the case on my RTX3080ti. I don't have experience with T4. I will check on a M40 if i have time.

**Comment by johnlee6511** [comment]
Thank you for your reply
I resolve it. swap model did not processed in GPU
I takes 2 seconds in T4 GPU

Then, Can you give me some tips to reduce the seconds?
Thanks
